cryptoki/src/functions/session_management.rs

```diff
@@ -42,20 +42,26 @@ impl<'a> Session<'a> {
         unsafe { Rv::from(get_pkcs11!(self.client(), C_CloseSession)(self.handle())).into_result() }
     }
 
-    /// Log a session in
-    ///
-    /// Do not fail if the user is already logged in. It happens if another session on the same slot
-    /// has already called the log in operation. Record the login call and only log out when there
-    /// aren't anymore sessions requiring log in state.
-    pub fn login(&self, user_type: UserType) -> Result<()> {
-        self.client().login(self, user_type)
+    /// Log a session in.
+    pub fn login(&self, user_type: UserType, pin: &str) -> Result<()> {
+        let pin_ptr = match pin.len() {
+            0 => std::ptr::null_mut(),
```

Comment:
Should we distinguish between no pin and a pin of size 0? As in between `pPin` with a valid address but the len being 0 and `pPin` with the NULL value.